blockchain/chain_sync/src/sync.rs

```diff
@@ -510,16 +513,13 @@ where
         Ok(fts)
     }
     // Block message validation checks
-    fn check_block_msgs(&self, block: Block, tip: &Tipset) -> Result<(), Error> {
-        let mut pub_keys = Vec::new();
-        let mut cids = Vec::new();
-        for m in block.bls_msgs() {
-            let pk = self
-                .state_manager
-                .get_bls_public_key(m.from(), tip.parent_state())?;
-            pub_keys.push(pk);
-            cids.push(m.cid()?.to_bytes());
-        }
+    fn check_block_msgs(
+        blockstore: &DB,
+        db: Arc<DB>,
```

Comment:
You don't need both, these are the same (a note about this is references are needed for things that are done syncronously and you only need to clone the `Arc<DB>` when the action needs to be done in another thread)